internal/tools/clickhouse_create_traces_table.sql

```diff
@@ -1,27 +1,33 @@
 CREATE TABLE base.traces (
-    `id` UUID,
     `chain_id` UInt256,
     `block_number` UInt256,
     `block_hash` FixedString(66),
     `block_timestamp` UInt64 CODEC(Delta, ZSTD),
     `transaction_hash` FixedString(66),
     `transaction_index` UInt64,
-    `call_type` String,
+    `subtraces` Int64,
+    `trace_address` Array(Int64),
+    `type` LowCardinality(String),
+    `call_type` LowCardinality(String),
     `error` Nullable(String),
     `from_address` FixedString(42),
     `to_address` FixedString(42),
-    `gas` UInt128,
-    `gas_used` UInt128,
+    `gas` UInt64,
+    `gas_used` UInt64,
     `input` String,
     `output` Nullable(String),
-    `subtraces` UInt64,
-    `trace_address` String,
-    `trace_type` String,
     `value` UInt256,
+    `author` Nullable(FixedString(42)),
+    `reward_type` LowCardinality(Nullable(String)),
+    `refund_address` Nullable(FixedString(42)),
     `is_deleted` UInt8 DEFAULT 0,
     `insert_timestamp` DateTime DEFAULT now(),
-    INDEX hash_idx transaction_hash TYPE bloom_filter GRANULARITY 1,
-    INDEX to_address_idx to_address TYPE bloom_filter GRANULARITY 1,
-    INDEX from_address_idx from_address TYPE bloom_filter GRANULARITY 1,
+    INDEX idx_block_timestamp block_timestamp TYPE minmax GRANULARITY 1,
+    INDEX idx_block_number block_number TYPE minmax GRANULARITY 1,
+    INDEX idx_block_hash block_hash TYPE bloom_filter GRANULARITY 1,
+    INDEX idx_from_address from_address TYPE bloom_filter GRANULARITY 1,
+    INDEX idx_to_address to_address TYPE bloom_filter GRANULARITY 1,
+    INDEX idx_type type TYPE bloom_filter GRANULARITY 1,
 ) ENGINE = ReplacingMergeTree(insert_timestamp, is_deleted)
-ORDER BY (block_number, id) SETTINGS allow_experimental_replacing_merge_with_cleanup = 1;
\ No newline at end of file
+ORDER BY (chain_id, transaction_hash, trace_address, block_hash)
```

Comment:
why block_hash and not block_number?

Reply:
same as before. It will persist data even in case of reorgs